### PR TITLE
Android & Plain Java Benchmarks; Method Count Report

### PR DESCRIPTION
**Pull request by m4dc4p**
Implemented benchmarks measuring crypto operations on Android
and using plain Java (server/desktop environment).

* Android benchmarks in `android-benchmark`. Run via Android
  Studio or `connectedDeviceTest` gradle command.
* Plain java benchmarks in `plain-benchmark`. Implemented using
  JMH; run via an executable JAR. For details, see `DEVELOPMENT.md`.
* Updated build to report method counts; added some benchmarks.

### CODE REVIEW

File: android-benchmark/src/androidTest/java/CryptoBenchmark.kt [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+    }
+  })
+
+  if (!latch2.await(5, TimeUnit.SECONDS))
+    throw Exception("Failed to create writer key in 5 seconds.")
+
+  return writer1!!
+}
+
+// From https://github.com/renatoathaydes/kotlin-hidden-costs-benchmark
```
Comment (gstro) [inline android-benchmark/src/androidTest/java/CryptoBenchmark.kt:55]: Nice find.

File: android-benchmark/src/androidTest/java/CryptoBenchmark.kt [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+
+  @Benchmark
+  open fun encrypt(reps: Int): BlackHole = blackhole.apply {
+    for(i in 0 until reps)
+      consume(client.encryptRecord(recordType, RecordData(mapOf(Pair("data", bytes))), null, key))
+  }
+
+  @Benchmark
+  open fun sign(reps: Int): BlackHole = blackhole.apply {
+    val localRecord = LocalRecord(mapOf(Pair("data", bytes)), LocalMeta(client.clientId(), client.clientId(), recordType, null))
```
Comment (gstro) [inline android-benchmark/src/androidTest/java/CryptoBenchmark.kt:174]: Is there not a concern for object creation costs skewing the benchmark?

**Review by gstro — APPROVED** [review]
Nice work!

File: android-benchmark/src/androidTest/java/CryptoBenchmark.kt (hunk 2 above)
Comment (m4dc4p) [reply android-benchmark/src/androidTest/java/CryptoBenchmark.kt:174]: The `reps` parameter is pretty high (1000+) so the object creation should be negligible.

**Review by ericmann — APPROVED** [review]

**Pull request merged**